legate/core/store.py

```diff
@@ -1233,6 +1233,14 @@ def project(self, dim: int, index: int) -> Store:
             If ``dim`` is not a valid dimension name or ``index`` is
             out of bounds
         """
+        # This is a workaround for NumPy's expected behavior when adding
+        # an integer to np.uint64, resulting in float64.
+        if type(index) is not int:
+            import numpy as np
+
+            assert np.issubdtype(type(index), np.integer)
+            index = int(index)
+
```

Comment:
the check itself is as expensive as the casting, so we better cast it to `int` all the time
```suggestion
        index = int(index)
```